src/session_base.cpp

```diff
@@ -409,6 +409,11 @@ void zmq::session_base_t::process_attach (i_engine *engine_)
         zmq_assert (!_pipe);
         _pipe = pipes[0];
 
+        //  The endpoints strings are not set on bind, set them here so that
+        //  events can use them.
+        pipes[0]->set_endpoint_pair (engine_->get_endpoint());
```

Comment:
@sigiesec - this does the trick to make the endpoint available. On bind, the pipe is not created for tcp/ipc, that's why it was not set before and we didn't see the address.

Reply:
Ah :) And on connect, the endpoint_pair is already set, and it is now set to the same value again?

Reply:
On the connect side the pipe is created immediately, so this branch is never ran